fairscale/experimental/wgit/repo.py

```diff
@@ -13,108 +13,156 @@
 
 
 class Repo:
-    def __init__(self, parent_dir: Path, init: bool = False) -> None:
-        """Features:
-        1. Create the wgit directory if it does not exist.
-        2. SHA1Store.init()
-        3. Create SHA1 .wgit/sha1_refs.json
-        3. Initialize a .git directory within the .wgit using `git init`.
-        4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_refs.json`
-        """
+    """
+    Represents the WeiGit repo for tracking neural network weights and their versions.
+
+    Args:
+        parent_dir (pathlib.Path, str)
+            The path to the parent directory where a weigit repo will be created. In the case a repo already exists, it will be wrapped with this class.
+        init (bool, optional)
+            - If ``True``, initializes a new WeiGit repo in the parent_dir. Initialization creates a `.wgit` directory within the <parent_dir>, triggers an initialization
+                of a sha1_store in the ./<parent_dir>/.wgit directory, and makes the ./<parent_dir>/.wgit a git repository through git initialization.
+            - If ``False``, a new WeiGit repo is not initialized and the existing repo is simply wrapped, populating the `wgit_parent` and other internal attributes.
+            - Default: False
+    Methods:
+        add (in_file_path: str)
+            adds a file to the sha1_store. If a new version of an existing file is added, updates the reference to the existing file.
+        commit(message: str)
+            Commits the staged changes to the repo
+        status(self)
+            Shows the state of the working tree.
+        log(self, file: str)
+            shows the weigit log of the commit history
+        checkout(self, sha1: str)
+            Checkout a previously commited version of the checkpoint
+        compression(self)
+            Not implemented
+        checkout_by_steps
+            Not implemented
+        path
+            Get path to the weigit
```

Comment:
no need to add method list here. It can be out of date quickly if we update the methods below.